src/dsp/rx_fft.cpp

```diff
@@ -105,8 +109,13 @@ void rx_fft_c::get_fft_data(std::complex<float>* fftPoints, unsigned int &fftSiz
         return;
     }
 
+    std::chrono::time_point<std::chrono::steady_clock> now = std::chrono::steady_clock::now();
+    std::chrono::duration<double> diff = now - d_lasttime;
+    d_lasttime = now;
+
     /* perform FFT */
-    do_fft(d_cbuf.linearize(), d_cbuf.size());  // FIXME: array_one() and two() may be faster
+    d_cbuf.erase_begin(std::min((unsigned int)(diff.count() * d_quadrate * 1.001), (unsigned int)d_cbuf.size() - d_fftsize));
+    do_fft(d_cbuf.linearize(), d_fftsize);  // FIXME: array_one() and two() may be faster
```

Comment:
With respect to performance, one possible issue is that increasing the circular buffer size also increases the cost of the `linearize()` here. I haven't found it to be a problem, but it would probably be worth testing performance at high samples rates with & without this change. It may also make sense to reduce the extra capacity in the circular buffer -- I arbitrarily chose one second, but I'm not sure whether there are scenarios where samples arrive in bursts that large. Perhaps it's not unrealistic for TCP connections though...

Reply:
I see that `linearize()` can be eliminated by replacing it with `array_one()` and `array_two()` so I'll take a stab at that.

Reply:
I added a second commit which removes the `linearize()` calls. There is no appreciable difference in performance, even for the largest FFT size, because they only accounted for a very small fraction of the execution time.